internal/logs/syslog_transport.go

```diff
@@ -0,0 +1,80 @@
+package logs
+
+import (
+	"errors"
+	"fmt"
+	"log/syslog"
+	"net"
+	"os"
+	"sync"
+	"time"
+)
+
+const (
+	priority = syslog.LOG_INFO
+)
+
+// We implement this way to force timestamp with the original ones.
+type syslogTransport struct {
+	server   string
+	hostname string
+	conn     net.Conn
+	lock     sync.Mutex // guards conn
+}
+
+func NewSyslogServer(server string) (*syslogTransport, error) {
+	syslogInstance := &syslogTransport{
+		server: server,
+		lock:   sync.Mutex{},
+	}
+	err := syslogInstance.Init()
+	if err != nil {
+		return nil, err
+	}
+	return syslogInstance, nil
+}
+
+func (s *syslogTransport) Init() error {
+	hostname, _ := os.Hostname()
+	var err error
+
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	s.hostname = hostname
+	s.conn, err = net.Dial("tcp", s.server)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+func (s *syslogTransport) WriTeLog(logline *LogEntry) error {
+
+	s.lock.Lock()
+	hostname := s.hostname
+	s.lock.Unlock()
+
+	logEntry := fmt.Sprintf("<%d>%s %s %s[%d]: %s\n",
+		priority, logline.Time.Format(time.RFC3339), hostname,
+		logline.workload, os.Getpid(), logline.String())
+	return s.writeAndRetry(logEntry, 10)
+}
+
+func (s *syslogTransport) writeAndRetry(data string, retryTimes int) error {
+	if retryTimes < 0 {
+		return errors.New("Cannot send message")
+	}
+
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	if s.conn != nil {
+		_, err := fmt.Fprintf(s.conn, "%s", data)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+	return s.writeAndRetry(data, retryTimes-1)
```

Comment:
Why recursively?

Reply:
It would be also good to pause between retries - exponential backoff maybe?

Reply:
fixed.